docs/articles/deployments.md

```diff
@@ -0,0 +1,237 @@
+
+## Deployment status
+In all the above cases Marathon will keep retrying to deploy the task indefinitely. During the deployment lifetime its status can be queried from the `/v2/deployments` [Rest API](api.md) endpoint. For example, let's change the above `/product/service/rails-app` definition and make it request 1000 CPUs using `"cpus": 1000`. Here, Marathon will have difficulties finding an agent with enough CPUs so that we have enough time to examine the deployment using `GET /v2/deployments` request. The result looks like:
+```
+[
+    {
+        "id": "3bcb15fc-737f-4a36-b185-2ae3a703239e",
+        "affectedApps": [
+            "/product/service/rails-app"
+        ],
+        "affectedPods": [],
+        "steps": [
+            {
+                "actions": [{
+                        "action": "StartApplication",
+                        "app": "/product/service/rails-app"
+                    }
+                ]
+            },
+            {
+                "actions": [{
+                        "action": "ScaleApplication",
+                        "app": "/product/service/rails-app"
+                    }
+                ]
+            }
+        ],
+        "currentActions": [
+            {
+                "action": "ScaleApplication",
+                "app": "/product/service/rails-app"
+            }
+        ],
+        "currentStep": 2,
+        "totalSteps": 2
+    }
+]
+```
+This endpoint returns a list of currently active deployments (some fields omitted for brevity). Interesting fields are:
+- `id` - this is the `deploymentId` from the original response
+- `affectedApps` - a deployment may actually "affect" (add, remove or modify) multiple service definitions (e.g. in a case of a group deployment)
+- `steps` - a series of actions to achieve the defined goal. This is being referred to as a [deployment plan](deployments.md#deployment-plan)
+- `currentAction` - currently executed action of the plan
+
+## Deployment plan
+In the above example `steps` field holds the deployment plan. Our deployment plan has two deployment steps: `StartApplication` and `ScaleApplication`. Currently, following deployment steps exist:
+
+- `StartApplication` - this step exists for backward compatibility reasons. Currently, it is doing nothing and is immediately successful
+- `ScaleApplication` - is responsible for starting the defined number of tasks and waiting for them to be up and running
+- `StopApplication` - stops given service by killing its tasks and waiting for the kill confirmation
+- `RestartApplication` - used when modifying an existing service definition and restarting its tasks (using the new definition)
+
+Steps are executed sequentially, however actions within one step (should there be more than one) are executed in parallel. Once all steps are executed successfully, the deployment is finished and removed from the state.
+
+## Canceling a deployment
```

Comment:
Nit: Do we want to capitalize headers? So `Canceling a Deployment`?

Reply:
I thought about it, but as far as the template goes, we don't. Should we change it? Opinions?